### DISCUSSION TITLE
Stopping SPEAK_START Command in MMDAgent-EX

### DISCUSSION THREAD

**Discussion by ed-morais** [mmdagent-ex/MMDAgent-EX#8] [Q&A] [answered]
Hello,

I’m currently using MMDAgent-EX and I’ve encountered a situation where I need to stop the voice playback with lip sync during playback, which is initiated with the SPEAK_START command.

I understand that the SOUND_STOP command is used to stop sounds started with the SOUND_START command. However, it seems there isn’t a direct command to stop the voice playback started with SPEAK_START.

Could you please provide some guidance on how to stop the voice playback with lip sync during playback? Is there a specific command or workaround for this situation?

Any help would be greatly appreciated.

**Comment by LeeAkinobu** [comment, marked as answer]
Thanks,

Sorry that SPEAK_STOP is not currently implemented yet.  The only workaround is to divide the audio data you want to use in small pieces (per phrases or per words) and make a playing loop in .fst, sending them in turn at receiving SPEAK_EVENT_STOP of the last one, so that you can stop sending it at any timing.

I now recognizd the necessity of SPEAK_STOP function, and want to implement it to stop playing SPEAK.  It will be implemented in the next release, perhaps within a month.

**Reply by ed-morais** [reply]
Thank you so much for the prompt reply! 
It's great to hear that it will be implemented in the next release. 
Looking forward to the update. :)

**Reply by LeeAkinobu** [reply]
Now SPEAK_STOP has been implemented and v1.0.2 is released.  Please use it.

Note that SPEAK_STOP always issues SPEAK_EVENT_STOP, even if no audio is playing at that time. 

Thanks,

**Reply by ed-morais** [reply]
Thank you so much! :)
